components/memory/db/events.go

```diff
@@ -0,0 +1,169 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/leptonai/gpud/log"
+	"github.com/leptonai/gpud/pkg/sqlite"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const (
+	TableNameMemoryEvents = "component_memory_events"
```

Comment:
it appears most event tables have the same schema. Can we make this package universal and used by all components that reads/writes an event table? only need to pass in a different table name, and everything else should be the same

Reply:
https://github.com/leptonai/gpud/pull/327 addresses this.